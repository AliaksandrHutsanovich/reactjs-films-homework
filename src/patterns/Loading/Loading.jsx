import React from 'react';
import { motion } from 'framer-motion';

import { Text } from '../../components';

import styles from './Loading.scss';

const Loading = () => (
  <div className={styles.wrapper}>
    <ul className={styles.wrapper__list}>
      <motion.li
        className={styles.wrapper__list_item}
        animate={{
          scale: [1, 2, 1, 1, 2, 1, 1, 1],
        }}
        transition={{
          duration: 2,
          times: [0, 0.143, 0.286, 0.429, 0.572, 0.715, 0.858, 1],
          loop: Infinity,
          ease: 'easeInOut',
        }}
      />
      <motion.li
        className={styles.wrapper__list_item}
        animate={{
          scale: [1, 1, 2, 1, 1, 2, 1, 1],
        }}
        transition={{
          duration: 2,
          times: [0, 0.143, 0.286, 0.429, 0.572, 0.715, 0.858, 1],
          loop: Infinity,
          ease: 'easeInOut',
        }}
      />
      <motion.li
        className={styles.wrapper__list_item}
        animate={{
          scale: [1, 1, 1, 2, 1, 1, 2, 1],
        }}
        transition={{
          duration: 2,
          times: [0, 0.143, 0.286, 0.429, 0.572, 0.715, 0.858, 1],
          loop: Infinity,
          ease: 'easeInOut',
        }}
      />
    </ul>
    <Text type="normal">LOADING</Text>
  </div>
);

export default Loading;
