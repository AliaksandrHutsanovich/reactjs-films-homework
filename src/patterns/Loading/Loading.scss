@font-face{
  font-family: 'HelveticaNeue';
  src: url('../../assets/fonts/HelveticaNeue.ttf') format('truetype');
}

.wrapper {
  display: flex;
  flex-direction: column;
  align-items: center;
  &__list {
    display: flex;
    justify-content: space-between;
    width: 40px;
    list-style-type: none;
    padding: 0;
  }

  &__list_item {
    background-color: rgb(10, 174, 228);
    border-radius: 4px;
    width: 7px;
    height: 7px;
  }

  &__text {
    text-align: center;
    color: white;
    font: normal 15px HelveticaNeue;
  }
}
