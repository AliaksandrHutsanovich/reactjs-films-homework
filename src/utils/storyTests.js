const storyTests = (stories, renderer) => {
  Object.keys(stories).forEach((storyKey) => {
    if (typeof stories[storyKey] !== 'function') {
      return;
    }
    it(`${storyKey} should be rendered properly`, () => {
      renderer.render(stories[storyKey]());
      const result = renderer.getRenderOutput();

      expect(result).toMatchSnapshot();
    });
  });
};

export default storyTests;
