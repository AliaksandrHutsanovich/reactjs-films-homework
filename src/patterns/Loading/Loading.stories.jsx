import React from 'react';
import Loading from './Loading';

export default {
  component: Loading,
  title: 'patterns/Loading',
};

export const loadingExample = () => (
  <Loading />
);
