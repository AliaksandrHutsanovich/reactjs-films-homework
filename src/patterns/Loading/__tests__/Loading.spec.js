import ShallowRenderer from 'react-test-renderer/shallow';
import storyTests from '../../../utils/storyTests';
import * as stories from '../Loading.stories';

describe('<Loading />', () => {
  const renderer = new ShallowRenderer();

  storyTests(stories, renderer);
});
