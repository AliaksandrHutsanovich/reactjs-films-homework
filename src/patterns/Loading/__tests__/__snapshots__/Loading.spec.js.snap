// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loading /> loadingExample should be rendered properly 1`] = `
<div
  className="wrapper"
>
  <ul
    className="wrapper__list"
  >
    <ForwardRef(MotionComponent)
      animate={
        Object {
          "scale": Array [
            1,
            2,
            1,
            1,
            2,
            1,
            1,
            1,
          ],
        }
      }
      className="wrapper__list_item"
      transition={
        Object {
          "duration": 2,
          "ease": "easeInOut",
          "loop": Infinity,
          "times": Array [
            0,
            0.143,
            0.286,
            0.429,
            0.572,
            0.715,
            0.858,
            1,
          ],
        }
      }
    />
    <ForwardRef(MotionComponent)
      animate={
        Object {
          "scale": Array [
            1,
            1,
            2,
            1,
            1,
            2,
            1,
            1,
          ],
        }
      }
      className="wrapper__list_item"
      transition={
        Object {
          "duration": 2,
          "ease": "easeInOut",
          "loop": Infinity,
          "times": Array [
            0,
            0.143,
            0.286,
            0.429,
            0.572,
            0.715,
            0.858,
            1,
          ],
        }
      }
    />
    <ForwardRef(MotionComponent)
      animate={
        Object {
          "scale": Array [
            1,
            1,
            1,
            2,
            1,
            1,
            2,
            1,
          ],
        }
      }
      className="wrapper__list_item"
      transition={
        Object {
          "duration": 2,
          "ease": "easeInOut",
          "loop": Infinity,
          "times": Array [
            0,
            0.143,
            0.286,
            0.429,
            0.572,
            0.715,
            0.858,
            1,
          ],
        }
      }
    />
  </ul>
  <UNDEFINED
    type="normal"
  >
    LOADING
  </UNDEFINED>
</div>
`;
